go/libraries/doltcore/mvdata/engine_table_writer.go

```diff
@@ -265,52 +265,70 @@ func (s *SqlEngineTableWriter) createOrEmptyTableIfNeeded() error {
 
 // createTable creates a table.
 func (s *SqlEngineTableWriter) createTable() error {
-	cr := plan.NewCreateTable(sql.UnresolvedDatabase(s.database), s.tableName, false, false, &plan.TableSpec{Schema: s.tableSchema})
-	analyzed, err := s.se.Analyze(s.sqlCtx, cr)
-	if err != nil {
-		return err
+	// TODO don't use internal interfaces to do this, we had to have a sql.Schema somewhere
+	// upstream to make the dolt schema
+	sqlCols := make([]string, len(s.tableSchema.Schema))
```

Comment:
Surely there's a library function that can generate a create table statement

If we need an engine to do that fine, but this likely to break as we add features

Reply:
I am concerned about all of the internal engine interface uses, but it didn't feel right to fix all of those in this PR. This change in particular seems pretty unnecessary because we do some BS to convert sql.Schema-> dolt Schema and then this hack to go backwards again.